### ISSUE TITLE
EventTarget's methods don't validate [this] arguments

### ISSUE THREAD

**Issue by Phosra** [denoland/deno#12666]
Currently, `EventTarget`'s prototype methods don't check the type (prototype) of the `this` argument passed to them.

E.g. given this code,
```js
const this_ = {};
const event_name = "";
const event_handler = null;
const options = {};
EventTarget.prototype.addEventListener.call(this_, event_name, event_handler, options);
```
Firefox and Chrome, respectively throw the following:
```
TypeError: 'addEventListener' called on an object that does not implement interface EventTarget.
```
```
TypeError: Illegal invocation
```

In contrast, on the current canary (1.15.3+26a5471), it does nothing, and returns `undefined`.

**Comment by lucacasonato** [comment]
I have a very old branch somewhere that fixes this. I'll assign it to myself - might get this fixed for 1.16.1.

**Comment by Garcia101** [comment]
> I have a very old branch somewhere that fixes this. I'll assign it to myself - might get this fixed for 1.16.1.

Any update on this?
I've since accidentally encountered another bug related to the code in question: #14254.
